tower-buffer/src/lib.rs

```diff
@@ -270,16 +308,18 @@ where
 impl<T, Request> Service<Request> for Buffer<T, Request>
 where
     T: Service<Request>,
+    <T as Service<Request>>::Error: Into<Error> + Send + Sync + 'static,
+    <T as Service<Request>>::Error: fmt::Display + fmt::Debug,
 {
     type Response = T::Response;
-    type Error = Error<T::Error>;
+    type Error = Error;
     type Future = ResponseFuture<T::Future, T::Error>;
 
     fn poll_ready(&mut self) -> Poll<(), Self::Error> {
         // If the inner service has errored, then we error here.
         self.tx
             .poll_ready()
-            .map_err(move |_| Error::Closed(self.get_error_on_closed()))
+            .map_err(move |_| ClosedError(self.get_error_on_closed()).into())
```

Comment:
TIOLI: It might be slightly tidier if you just modified `get_error_on_closed` to return a `ClosedError<T::Error>`, rather than having to wrap it here?